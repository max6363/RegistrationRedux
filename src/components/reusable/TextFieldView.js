import React, {Component} from 'react';
import {View, TextInput, Text, TouchableOpacity, Image} from 'react-native';

export default class TextFieldView extends Component {
  constructor(props) {
    super(props);
  }
  renderShowHideButton() {
    const {showHideButtonStyle} = styles;
    const {isSecureTextVisible} = this.props;
    const icon =
      isSecureTextVisible === true
        ? require('./../img/unmask.png')
        : require('./../img/mask.png');
    if (this.props.isSecureTextField) {
      return (
        <TouchableOpacity
          onPress={this.props.onPasswordShowHideButtonPress}
          style={showHideButtonStyle}
          testID={this.props.passwordShowHideTestID}
          accessibilityRole={'button'}
          accessibilityLabel={this.props.passwordShowHideTestID}
          accessible={true}>
          <Image source={icon} resizeMode="contain" />
        </TouchableOpacity>
      );
    }
  }
  render() {
    const {container, labelStyle, textStyle, errorLabelStyle} = styles;
    const {label, text, textInputTestID} = this.props;
    const error = this.props.error();

    return (
      <View style={container}>
        <View style={labelStyle.container}>
          <Text style={labelStyle.text}>{label}</Text>
        </View>
        <View style={textStyle.container}>
          <TextInput
            testID={textInputTestID}
            style={textStyle.text}
            secureTextEntry={
              this.props.isSecure
                ? this.props.isSecureTextVisible
                  ? false
                  : true
                : false
            }
            onChangeText={this.props.onChangeText}
            onBlur={this.props.onBlur}
            onFocus={this.props.onFocus}>
            {text}
          </TextInput>
          {this.renderShowHideButton()}
        </View>
        {error !== null && error !== undefined ? (
          <View style={errorLabelStyle.container}>
            <Text style={errorLabelStyle.text}>{error}</Text>
          </View>
        ) : null}
      </View>
    );
  }
}

const styles = {
  container: {
    flex: 1,
    alignSelf: 'stretch',
    marginTop: 10,
    marginBottom: 10,
    marginLeft: 15,
    marginRight: 15,
  },
  labelStyle: {
    container: {
      alignSelf: 'stretch',
    },
    text: {fontSize: 15},
  },
  textStyle: {
    container: {
      alignSelf: 'stretch',
      alignItems: 'center',
      justifyContent: 'center',
      borderWidth: 1,
      borderColor: '#00000090',
      borderRadius: 4,
      marginTop: 5,
      padding: 5,
      flexDirection: 'row',
    },
    text: {fontSize: 22, flex: 1},
  },
  errorLabelStyle: {
    container: {
      alignSelf: 'stretch',
      marginTop: 5,
    },
    text: {fontSize: 11, color: 'red'},
  },
  showHideButtonStyle: {
    flex: 1.2,
    alignSelf: 'center',
    alignItems: 'flex-end',
    justifyContent: 'center',
    marginRight: 5,
  },
};
