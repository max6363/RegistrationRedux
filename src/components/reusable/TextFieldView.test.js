import React from 'react';
import {shallow} from 'enzyme';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({adapter: new Adapter()});

import TextFieldView from './TextFieldView';

describe('Test TextFieldView', () => {
  it('can test TextFieldView with props', () => {
    const wrapper = shallow(
      <TextFieldView
        label="Label"
        text="Text"
        error={() => {
          return 'error message';
        }}
        onBlur={() => {}}
        onFocus={() => {}}
        onChangeText={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('can test TextFieldView with props (error is null)', () => {
    const wrapper = shallow(
      <TextFieldView
        label="Label"
        text="Text"
        error={() => {
          return null;
        }}
        onBlur={() => {}}
        onFocus={() => {}}
        onChangeText={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
