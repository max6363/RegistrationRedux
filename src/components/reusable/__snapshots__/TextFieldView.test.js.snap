// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextFieldView can test TextFieldView with props (error is null) 1`] = `ShallowWrapper {}`;

exports[`Test TextFieldView can test TextFieldView with props 1`] = `ShallowWrapper {}`;
